Marlin/src/gcode/calibrate/G28.cpp

```diff
@@ -328,6 +334,22 @@ void GcodeSuite::G28(const bool always_home_all) {
         delayed_move_time = 0;
         active_extruder_parked = true;
 
+        if (IDEX_saved_mode >= DXC_DUPLICATION_MODE) {
+          enqueue_and_echo_commands_now_P(PSTR("M605 S1\nT0"));
+          char cmd[16];
+          sprintf_P(cmd, PSTR("M605 S2 X%i"), DEFAULT_DUPLICATION_X_OFFSET);
+          enqueue_and_echo_command_now(cmd);
+          enqueue_and_echo_commands_now_P(PSTR("G28 X"));
+          sprintf_P(cmd, PSTR("G1 X%i"), (X_BED_SIZE) / 3);
+          enqueue_and_echo_command_now(cmd);
+          if (IDEX_saved_mode >= DXC_SCALED_DUPLICATION_MODE) {
+            sprintf_P(cmd, PSTR("M605 S3 X%i"), DEFAULT_DUPLICATION_X_OFFSET);
+            enqueue_and_echo_commands_now_P(cmd);
+          }
+        }
```

Comment:
The way the Marlin G-code queue works, these are going to cause trouble. At the time when `GcodeSuite::G28` is called, the first-out command in the command queue is likely `G28`. The queue is blocked now and no more commands can go out of the queue until this returns back to the main loop, which then resumes processing the G-code command queue. If the queue is full at this time, Marlin will sit in an infinite loop, stuck here because the queue is never getting emptied.

Therefore, we cannot invoke G-code commands from within other G-code commands. Instead, you must call functions that do what `M605 S1`, `T0`, `M605 S2 Xn`, etc. do. And clearly it's a bad idea to invoke `G28` from within `G28`.

So, this must all be rewritten.

Reply:
I'm extending the `USE_EXECUTE_COMMANDS_IMMEDIATE` stuff so that it will be possible to run commands from within other commands, just as though they were functions. It's a useful bit of voodoo added by @marcio-ao a while ago. So far it's only been used with `G29_RETRY_AND_RECOVER`, but should be applicable to any situation where it would otherwise require turning G-code handlers into functions. I'll post the update as a PR for you to check out.